### PR TITLE
tests: add test suite & CI

### PR DESCRIPTION
**Pull request by RyanZim**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+let store
+
+beforeEach(() => {
+  store = Redux.createStore(reducer)
+})
+
+test('awaitState waits for state to update', async () => {
+  // queue up actions
+  setTimeout(() => store.dispatch({ type: 'TOGGLE_ENABLED' }), 1000)
```
Comment (mvayngrib) [inline test.js]: nit: do these need to be this long? Or maybe we can use fake timers?

**Review by mvayngrib — APPROVED** [review]
utACK

File: test.js (hunk 1 above)
Comment (RyanZim) [reply test.js]: Probably not; they're just there to ensure the event loop clears; I don't know how short you could make them until you run into trouble. Also serve a secondary purpose, as you can now manually verify test run times look correct, to ensure it's actually waiting on the action.

Comment (mvayngrib) [reply test.js]: what do u think of this approach with fake timers?
```diff
diff --git a/test.js b/test.js
index 7569759..62ea483 100644
--- a/test.js
+++ b/test.js
@@ -4,6 +4,8 @@ const awaitState = require('.')
 
 const initialState = { enabled: false }
 
+jest.useFakeTimers()
+
 function reducer(state = initialState, action) {
   switch (action.type) {
     case 'TOGGLE_ENABLED':
@@ -29,7 +31,9 @@ test('awaitState waits for state to update', async () => {
   setTimeout(() => store.dispatch({ type: 'TOGGLE_ENABLED' }), 1000)
 
   expect(store.getState().enabled).toBe(false)
-  await awaitState(store, isEnabledSelector)
+  const statePromise = awaitState(store, isEnabledSelector)
+  jest.runAllTimers()
+  await statePromise
   expect(store.getState().enabled).toBe(true)
 })
 
```

Comment (RyanZim) [reply test.js]: There's no point in using fake timers; the only purpose of timers is to delay stuff beyond the end of the event loop.

Comment (mvayngrib) [reply test.js]: the point is to make the tests run fast. Why not use nextTick if that's the only purpose?

Comment (RyanZim) [reply test.js]: `nextTick` might work; I just don't know the internals of redux, if they're not fully synchronous, you no longer have guaranteed execution order.

Comment (mvayngrib) [reply test.js]: they are fully synchronous

**Review by mvayngrib — APPROVED** [review]
utACK

**Pull request merged**